extras/videoDrivers/WinAPI/WinAPI.cpp

```diff
@@ -253,7 +254,7 @@ RECT VideoWinAPI::CalculateWindowRect(bool fullscreen, VideoMode& size) const
     return wRect;
 }
 
-bool VideoWinAPI::RegisterAndCreateWindow(const std::string& title, const VideoMode& wndSize, bool fullscreen)
+bool VideoWinAPI::RegisterAndCreateWindow(const std::string& title, const VideoMode& wndSize, WindowMode windowMode)
```

Comment:
I think you accidently removed `bool fullscreen` here as last parameter

Reply:
Yes, thanks! I did see the AppVeyor failure. I had initially folded the `fullscreen` parameter into the `enum` and realized later that this would lose information. I'm still considering renaming the `enum` to `WindowFlags` and turning it into a flag `enum` (I remember seeing a convenience macro elsewhere).
That way, we could extend it later to add – for example – borderless fullscreen if anyone ever wanted to do that.

Reply:
A quick image search shows this setting is typically called "Display Mode". A `DisplayMode` flag `enum` would allow folding the `fullscreen` parameter back into the `enum`. Settings would still store one `bool` per bit for compatibility.

Reply:
I think that is a good idea, as I don't like having (only) `WindowMode::Resizable` vs `WindowMode::NonResizable`, i.e. a boolean enum with a non-obvious type name. Hence I already wanted to suggest `WindowResizable::Yes/No` instead. But a flags-type is better I guess.

SDL calls it `WindowFlags`. I like having the "Flags" in the name so it is clear that this isn't meant to be used directly. Not sure how other projects usually call it, but my tendency would be rather to have "Window" in the name than "Display" as to me it wouldn't be obvious what "Display" refers to in a window creation function. That notwithstanding: If there is some commonly used name we should use that too.

Reply:
I looked at `enumUtils.h` and found it lacking. For example, it implements only `operator&()` and `operator|()`. I've extended it a bit and put all the code at the start of `VideoInterface.h` **for illustration purposes only**. Obviously, that would go into a libutil PR and the `enumUtils.h` header.

I'd like a thumbs up or thumbs down on the `enum` stuff before I adapt the rest of the code and work on that libutil PR.

I've at least one other issue once that is settled and then we can talk about the name.

Reply:
Oh, the unary `~` is of course a bad idea – circumvents type safety. Would be moved into utility functions.